src/geode/model/helpers/component_mesh_queries.cpp

```diff
@@ -137,4 +162,234 @@ namespace geode
             "triangles and quadrangles." );
         return {};
     }
+
+    std::pair< absl::InlinedVector< geode::PolygonVertices, 2 >,
+        absl::InlinedVector< geode::PolygonVertices, 2 > >
+        polygon_vertices_to_block_facets_vertices( const geode::BRep& model,
+            const geode::Block3D& block,
+            const geode::PolygonVertices& polygon_unique_vertices )
+    {
+        const auto polyhedra_around_polygon_facet =
+            block_mesh_polyhedra_from_unique_vertices_facet(
+                model, block, polygon_unique_vertices );
+
+        const auto nb_vertices = polygon_unique_vertices.size();
+        absl::InlinedVector< geode::PolygonVertices, 2 > facets_block_vertices;
+        absl::InlinedVector< geode::PolygonVertices, 2 > facets_unique_vertices;
+        const auto& block_mesh = block.mesh();
+        for( const auto facet_id :
+            geode::LRange{ polyhedra_around_polygon_facet.size() } )
+        {
+            geode::PolygonVertices f_block_vertices( nb_vertices );
+            geode::PolygonVertices f_unique_vertices( nb_vertices );
+            for( const auto facet_vertex_id : geode::LRange{ nb_vertices } )
+            {
+                f_block_vertices[facet_vertex_id] =
+                    block_mesh.polyhedron_facet_vertex(
+                        { polyhedra_around_polygon_facet[facet_id],
+                            facet_vertex_id } );
+                f_unique_vertices[facet_vertex_id] =
+                    model.unique_vertex( { block.component_id(),
+                        f_block_vertices[facet_vertex_id] } );
+            }
+            facets_block_vertices.emplace_back( std::move( f_block_vertices ) );
+            facets_unique_vertices.emplace_back(
+                std::move( f_unique_vertices ) );
+        }
+        return { facets_block_vertices, facets_unique_vertices };
+    }
+
+    /*!
+     * Returns the result of polygon_vertices_to_block_facets_vertices, but with
+     * the lists oriented so that the first PolygonVertices of each vector is
+     * linked to the facet with the same orientation as the given vertices.
+     */
+    std::tuple< absl::InlinedVector< geode::PolygonVertices, 2 >,
+        absl::InlinedVector< geode::PolygonVertices, 2 >,
+        bool >
```

Comment:
maybe i am wrong, but why do not you use `BlockPolyhedraFacetVertices` as return type?

Reply:
Yes, they're not the same object, this function returns two lists: the PolygonVertices representing the vertices in the block of each facet, and the PolygonVertices rerpesenting the unique vertices associated (with a corresponding numbering and orientation in the two lists.